### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by IdanTo1**

### CODE REVIEW

File: src/asm_task2Assignment1.s [hunk 1]
```diff
@@ -3,6 +3,7 @@ section	.rodata			; we define (global) read-only variables in .rodata section
 
 section .bss			; we define (global) uninitialized variables in .bss section
 	an: resb 12		; enough to store integer in [-2,147,483,648 (-2^31) : 2,147,483,647 (2^31-1)]
+	an_rev: resb 12		; enough to store integer in [-2,147,483,648 (-2^31) : 2,147,483,647 (2^31-1)]
 
 section .data			; we define (global) ininitialized variables in .data section
 	number: dd 0			; the int value will be at most max int, and so dword is enough
```
Comment (noakrams) [inline src/asm_task2Assignment1.s:9]: I don't think you used number

File: src/asm_task2Assignment1.s [hunk 2]
```diff
@@ -18,7 +19,7 @@ convertor:
 	mov ecx, dword [ebp+8]	; get function argument (pointer to string)
 
 	mov eax, 0					; reset number value to 0
-	calc_number:
+	calc_number:				; extract the decimal/binary value of the number from the digits string
 		cmp byte[ecx], 0x0 		; check if the next character is zero (i.e. null string termination)
 		jz calc_hex_init		; if finished to update number, go to next step
```
Comment (noakrams) [inline src/asm_task2Assignment1.s:24]: didn't understand why check both new line and null, maybe just check for a new line is enoufh? :)

Comment (IdanTo1) [reply src/asm_task2Assignment1.s:24]: I thought like you in the beginning, but they specifically said check them both, and I think I found a case where no \n present. I'll check it and update

File: src/asm_task2Assignment1.s (hunk 1 above)
Comment (IdanTo1) [reply src/asm_task2Assignment1.s:9]: Absolutely! Thanks 😅

File: src/asm_task2Assignment1.s (hunk 2 above)
Comment (IdanTo1) [reply src/asm_task2Assignment1.s:22]: check out this case
`bin/task2.bin < test.txt`
when test.txt is a file without \n

Comment (noakrams) [reply src/asm_task2Assignment1.s:22]: nice, I didn't think about it :)

**Pull request merged**